prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm.c

```diff
@@ -526,7 +532,11 @@ fi_ibv_rdm_tagged_got_recv_completion(struct fi_ibv_rdm_ep *ep,
 	struct fi_ibv_rdm_tagged_buf *rbuf =
 	    fi_ibv_rdm_tagged_get_rbuf(conn, ep, imm_data);
 	fi_ibv_rdm_tagged_process_recv(ep, conn, arrived_len, imm_data, rbuf);
-	fi_ibv_rdm_tagged_release_remote_sbuff(conn, ep, imm_data);
+
+	if ((imm_data % ep->n_buffs) == 0)
```

Comment:
Probably check rbuff == fi_ibv_rdm_tagged_get_rbuf(0) will be better because imm is not supported everywhere and can be dropped here.